nikola/image_processing.py

```diff
@@ -81,111 +81,132 @@ def filter_exif(self, exif, whitelist):
+        """Make a copy of the image in the requested size(s).
+
+        max_sizes should be a list of sizes, and the image would be resized to fit in a
+        square of each size (preserving aspect ratio).
+
+        dst_paths is a list of the destination paths, and should be the same length as max_sizes.
+
+        Backwards compatibility:
+
+        * If max_sizes is None, it's set to [max_size]
+        * If dst_paths is None, it's set to [dst]
+        * Either max_size or max_sizes should be set
+        * Either dst or dst_paths should be set
+        """
+        if dst_paths is None:
+            dst_paths = [dst]
+        if max_sizes is None:
+            max_sizes = [max_size]
+        if len(max_sizes) != len(dst_paths):
+            raise ValueError('resize_image called with incompatible arguments: {} / {}'.format(dst_paths, max_sizes))
         extension = os.path.splitext(src)[1].lower()
         if extension in {'.svg', '.svgz'}:
-            self.resize_svg(src, dst, max_size, bigger_panoramas)
+            self.resize_svg(src, dst_paths, max_sizes, bigger_panoramas)
             return
 
-        im = Image.open(src)
+        _im = Image.open(src)
 
         # The jpg exclusion is Issue #3332
-        if hasattr(im, 'n_frames') and im.n_frames > 1 and extension not in {'.jpg', '.jpeg'}:
-            # Animated gif, leave as-is
-            utils.copy_file(src, dst)
-            return
-
-        size = w, h = im.size
-        if w > max_size or h > max_size:
-            size = max_size, max_size
-
-            # Panoramas get larger thumbnails because they look *awful*
-            if bigger_panoramas and w > 2 * h:
-                size = min(w, max_size * 4), min(w, max_size * 4)
+        is_animated = hasattr(_im, 'n_frames') and _im.n_frames > 1 and extension not in {'.jpg', '.jpeg'}
 
         try:
-            exif = piexif.load(im.info["exif"])
-        except KeyError:
-            exif = None
-        # Inside this if, we can manipulate exif as much as
-        # we want/need and it will be preserved if required
-        if exif is not None:
+            exif = piexif.load(_im.info["exif"])
             # Rotate according to EXIF
             value = exif['0th'].get(piexif.ImageIFD.Orientation, 1)
             if value in (3, 4):
-                im = im.transpose(Image.ROTATE_180)
+                _im = _im.transpose(Image.ROTATE_180)
             elif value in (5, 6):
-                im = im.transpose(Image.ROTATE_270)
+                _im = _im.transpose(Image.ROTATE_270)
             elif value in (7, 8):
-                im = im.transpose(Image.ROTATE_90)
+                _im = _im.transpose(Image.ROTATE_90)
             if value in (2, 4, 5, 7):
-                im = im.transpose(Image.FLIP_LEFT_RIGHT)
+                _im = _im.transpose(Image.FLIP_LEFT_RIGHT)
             exif['0th'][piexif.ImageIFD.Orientation] = 1
+            exif = self.filter_exif(exif, exif_whitelist)
+        except KeyError:
```

Comment:
Is that KeyError meant to be coming from `_im.info["exif"]`? We could replace it with an `'exif' in _im.info` or `_im.info.get('exif')`. Same applies to `exif['0th']`.

Reply:
Probably (who can remember? not me :-D) will change.

Reply:
Done